testnet/stacks-node/src/tests/signer.rs

```diff
@@ -70,98 +71,120 @@ struct RunningNodes {
-        // Spawn all signers before the node to ensure their listening ports are open for the node event observer to bind to
-        for i in (0..num_signers).rev() {
+        for i in 0..num_signers {
             let (cmd_send, cmd_recv) = channel();
             let (res_send, res_recv) = channel();
             info!("spawn signer");
-            running_signers.insert(
-                i,
-                spawn_signer(&signer_configs[i as usize], cmd_recv, res_send),
-            );
-            signer_cmd_senders.insert(i, cmd_send);
+            running_signers.push(spawn_signer(
+                &signer_configs[i as usize],
+                cmd_recv,
+                res_send,
+            ));
+            signer_cmd_senders.push(cmd_send);
             result_receivers.push(res_recv);
         }
 
         // Setup the nodes and deploy the contract to it
-        let node = setup_stx_btc_node(
-            naka_conf,
-            num_signers,
-            &signer_stacks_private_keys,
-            &signer_configs,
-        );
-
-        // Calculate which signer will be selected as the coordinator
-        let config = stacks_signer::config::Config::load_from_str(&signer_configs[0]).unwrap();
+        let node = setup_stx_btc_node(naka_conf, &signer_stacks_private_keys, &signer_configs);
+        let config = SignerConfig::load_from_str(&signer_configs[0]).unwrap();
         let stacks_client = StacksClient::from(&config);
-        let (coordinator_id, coordinator_pk) =
-            calculate_coordinator(&config.signer_ids_public_keys, &stacks_client);
-        info!(
-            "Selected coordinator id: {:?} with pk: {:?}",
-            &coordinator_id, &coordinator_pk
-        );
-
-        // Fetch the selected coordinator and its cmd_sender
-        let running_coordinator = running_signers
-            .remove(&coordinator_id)
-            .expect("Coordinator not found");
-        let coordinator_cmd_sender = signer_cmd_senders
-            .remove(&coordinator_id)
-            .expect("Command sender not found");
 
         Self {
             running_nodes: node,
             result_receivers,
-            _signer_cmd_senders: signer_cmd_senders,
-            coordinator_cmd_sender,
-            running_coordinator,
+            signer_cmd_senders,
             running_signers,
             signer_stacks_private_keys,
             stacks_client,
         }
     }
 
+    fn get_current_reward_cycle(&self) -> u64 {
+        let block_height = self
+            .running_nodes
+            .btc_regtest_controller
+            .get_headers_height();
+        self.running_nodes
+            .btc_regtest_controller
+            .get_burnchain()
+            .block_height_to_reward_cycle(block_height)
+            .unwrap()
+    }
+
+    // Will panic if called on a reward cycle that has not had its signers calculated yet
+    fn get_coordinator_sender(&self, reward_cycle: u64) -> &Sender<RunLoopCommand> {
```

Comment:
@obycode this is where I use get_registered_signers_info to get the registerd signer ids and such. Combine this with where I tag you next...